pkg/provider/gitlab/gitlab.go

```diff
@@ -155,10 +156,28 @@ func (g *Gitlab) NewClient(ctx context.Context, store esv1beta1.GenericStore, ku
 	return g, nil
 }
 
-// Empty GetAllSecrets.
+// GetAllSecrets syncs all gitlab project variables into a single Kubernetes Secret.
 func (g *Gitlab) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+	if utils.IsNil(g.client) {
+		return nil, fmt.Errorf(errUninitializedGitlabProvider)
+	}
+
+	allData, _, err := g.client.ListVariables(g.projectID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	checkName := ref.Name != nil && len(ref.Name.RegExp) > 0
+	secretData := make(map[string][]byte)
+	for _, data := range allData {
+		ok, secretName := matchesFilter(checkName, ref, g.environment, data)
+		if !ok {
+			continue
+		}
+		secretData[secretName] = []byte(data.Value)
+	}
+
+	return secretData, nil
```

Comment:
does gitlab support find by tags? If so, can we add it to this implementation?

Reply:
Nope. The concept of tags for variables does not exist.